### DISCUSSION TITLE
Discussion about Implementing `MakeFile` in Open-Horizon docs repo

### DISCUSSION THREAD

**Discussion by Abhijay007** [open-horizon/open-horizon.github.io#191] [Ideas]
This discussion is meant to capture issues, questions, and requirements related to the implementation of the `MakeFile` in open-horizon docs. 

To give you all an overview about this and what the `make` command does and how it can be helpful for us?  I opened issue #189 

@joewxboy suggested that implementing ` MakeFile` will also allow us to create an `init` or similar target for installation, and to implement different logic for the different OSes. 

You all can also share your thoughts on this and let us know your views on it :)

**Comment by joewxboy** [comment]
Let's outline the Makefile targets.  We can start by borrowing from the `open-horizon-services` [Best Practices](https://github.com/open-horizon-services/Getting-Started#makefile-targets-for-microservices):

- `init` - any "run once" tasks go here: installing pre-requisites, ensuring all tools are up-to-date, updating configuration
- `build` - build the site locally once
- `dev` - build the site, start the web server, auto-build when changes are detected
- `run` - run the web server, no auto-build
- <strike>`stop`</strike> - stop the web server (is this needed?)
- <strike>`agent-run`</strike>
- <strike>`agent-stop`</strike>
- `test` - build once and look for errors and warnings (how to expose?)
- <strike>`push` </strike>- add files, commit with sign-off, push branch to forked repo
- <strike>`publish`</strike>
- <strike>`clean`</strike> or give instructions on how to delete the local clone?
- <strike>`distclean`</strike>

Thoughts?

**Reply by Abhijay007** [reply]
@joewxboy nice, all the other targets are seems fine to me but I think `stop`  and `push` is not needed, and for `push`  instead to doing this it is better to define an alias in `~/.gitconfig` file like this : 

```
[alias]
  amend = commit -s --amend
  cm = commit -s -m
  commit = commit -s
  ```
users may add this globally so that they don't need to add signoff again and again

**Reply by joewxboy** [reply]
Nice additions.  I didn't know about ~/.gitconfig

Time for me to read up on this.

**Comment by joewxboy** [comment]
@Abhijay007 Your `.gitconfig` additions are a great way to help new contributors to automagically sign-off on their commits.  Can we add your "define an alias" instructions someplace?  Maybe open an issue to add it to [this page on the documentation web site](https://open-horizon.github.io/common-requests/contribute.html).

**Reply by Abhijay007** [reply]
Ya sure, we can also take reference from [here](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits) for the instructions , will do that and open an issue for this :)

**Reply by joewxboy** [reply]
Possibly.  Make sure that their documentation has a license that would permit our usage (like CC-by-4.0).

**Reply by Abhijay007** [reply]
sure

**Reply by Abhijay007** [reply]
They are using Apache license 2.0 which allows users to use the software for any purpose, to distribute it, to modify it, and to distribute modified versions of the software under the terms of the license, without concern for royalties, so we probably use that info according to my research, you can read more about this license [here](https://en.wikipedia.org/wiki/Apache_License) :)

**Comment by Ruchip16** [comment]
Great points @Abhijay007 if this gets implemented various issues like signing off git commits, automating the build run tasks would get easier.

**Comment by joewxboy** [comment]
It appears that the technique we were planning to use for colorizing output is not supported on all OSes, so we'll drop that for now.  I believe that @Abhijay007 is still looking for others to help test.
